porcupine/tabs.py

```diff
@@ -386,6 +389,56 @@ class ReloadInfo(utils.EventDataclass):
     was_modified: bool
 
 
+def _ask_encoding(path: pathlib.Path, encoding_that_didnt_work: str) -> str | None:
+    dialog = tkinter.Toplevel()
+    big_frame = ttk.Frame(dialog)
+    big_frame.pack(fill="both", expand=True)
+    ttk.Label(
+        big_frame,
+        text=f'The content of "{path}" is not valid utf-8. Choose an encoding to use instead:',
```

Comment:
It should also mention how to change default encoding, in case you need to work on a legacy project. But because non-utf8 is project-specific weirdness, it really shouldn't be possible to specify an encoding in `filetypes.toml` and that feature should just be deleted.